src/evo/specialtx.cpp

```diff
@@ -85,6 +86,10 @@ bool CheckSpecialTxNoContext(const CTransaction& tx, CValidationState& state)
             // provider-update-revoke
             return CheckProUpRevTx(tx, nullptr, state);
         }
+        case CTransaction::TxType::LLMQCOMM: {
+            // quorum commitment
+            return llmq::CheckLLMQCommitment(tx, nullptr, state);
+        }
```

Comment:
quick idea about this:
What if we decouple this entire switch into its own function:
``` 
bool CheckTxByPayloadType(const CTransaction& tx, CBlockIndex* pindex, CValidationState& state)
```
 So we can call to the same function from `CheckSpecialTxNoContext` and from `CheckSpecialTx` and stop worrying about having to duplicate the code in both places.

Reply:
good idea. added in 344990e4dc75361e9afe364ee0ed3ff5cb65440a